app/controllers/articles_controller.rb

```diff
@@ -12,22 +13,34 @@ def create
   end
 
   def show
-    @article = Article.find_by(id: params[:id])
+    @article
   end
 
   def destroy
-    article = Article.find_by(id: params[:id])
-    article.destroy
-    flash[:success] = "#{article.title} has been deleted"
+    @article
+    @article.destroy
+    flash[:success] = "#{@article.title} has been deleted"
     redirect_to root_path
   end
 
+  def update
+    @article.update(article_params) ? (redirect_to article_path(@article)) : (render 'edit')
+  end
+
+  def edit
+    @article
```

Comment:
There's no need for this line. The article is available in this action due to the `before_action`